modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/helper/SwipeButtonLayout.kt

```diff
@@ -0,0 +1,216 @@
+                episode = baseEpisode,
+                swipeSource = swipeSource,
+            )
+            onItemUpdated(baseEpisode, rowIndex)
+        }
+    }
+
+    class DeleteFileButton(
+        private val onItemModified: (UserEpisode, RowIndex) -> Unit,
+        private val swipeSource: EpisodeItemTouchHelper.SwipeSource,
+        private val fragmentManager: FragmentManager,
+        private val viewModel: SwipeButtonLayoutViewModel,
+    ) : SwipeButton {
+
+        override val iconRes = VR.drawable.ic_delete
+
+        override val backgroundColor: (Context) -> Int =
+            { it.getThemeColor(UR.attr.support_05) }
+
+        override val onClick: (BaseEpisode, RowIndex) -> Unit = { episode, rowIndex ->
+            if (episode !is UserEpisode) {
+                throw IllegalStateException("Can only delete user episodes, but tried to delete: $episode")
+            }
+            viewModel.deleteEpisode(
+                episode = episode,
+                swipeSource = swipeSource,
+                fragmentManager = fragmentManager,
+                onDismiss = { onItemModified(episode, rowIndex) }
+            )
+        }
+    }
+
+    class ArchiveButton(
+        private val episode: BaseEpisode,
+        private val onItemUpdated: (BaseEpisode, RowIndex) -> Unit,
+        private val swipeSource: EpisodeItemTouchHelper.SwipeSource,
+        private val viewModel: SwipeButtonLayoutViewModel,
+    ) : SwipeButton {
+
+        override val iconRes
+            get() = if (episode.isArchived) {
+                IR.drawable.ic_unarchive
+            } else {
+                IR.drawable.ic_archive
+            }
+
+        override val backgroundColor: (Context) -> Int =
+            { it.getThemeColor(UR.attr.support_06) }
+
+        override val onClick: (BaseEpisode, RowIndex) -> Unit = { episode, rowIndex ->
+            if (episode !is PodcastEpisode) {
+                throw IllegalStateException("Can only share podcast episodes, but tried to archive: $episode")
+            }
+            viewModel.updateArchive(episode, swipeSource)
+            onItemUpdated(episode, rowIndex)
+        }
+    }
+
+    class ShareButton(
+        swipeSource: EpisodeItemTouchHelper.SwipeSource,
+        fragmentManager: FragmentManager,
+        context: Context,
+        viewModel: SwipeButtonLayoutViewModel,
+    ) : SwipeButton {
+
+        override val iconRes = IR.drawable.ic_share
+
+        override val backgroundColor: (Context) -> Int =
+            { it.getThemeColor(UR.attr.support_01) }
+
+        override val onClick: (BaseEpisode, RowIndex) -> Unit = { episode, _ ->
+            if (episode !is PodcastEpisode) {
+                throw IllegalStateException("Can only share podcast episodes: $episode")
+            }
+            viewModel.share(episode, fragmentManager, context, swipeSource)
+        }
+    }
+}
+
+/**
+ * - [onDeleteOrArchiveClick] will delete [UserEpisode]s and archive [PodcastEpisode]s
```

Comment:
Is this right? `onDeleteOrArchiveClick` I can't see it anywhere else in the code.